### DISCUSSION TITLE
CLPZ: "module file not found"

### DISCUSSION THREAD

**Discussion by haijinSk** [trealla-prolog/trealla-prolog#343] [Q&A] [answered]
Is it intentional that the precompiled, compact version/distribution of Trealla from "tpl-linux-x64.zip" does not include clpz (and clpb) module? 

From a user point of view:

```
?- use_module(library(clpz)).
Warning: module file not found: library/../library/clpz
   false.
```

But even if I have the clpz module file in the library folder alongside tpl executable, tpl can't see the module. Unless I run tpl with explicit path to tpl executable on the command line (or if I open terminal in the folder with tpl). And this is the case even if I compile/make the tpl from the source.

Maybe the problem is on my side. Not a Linux "power user", but I'm on Ubuntu, and I have .bashrc file with the path to the folder with tpl executable and there is no problem with other modules in this "loading library" context.

Anyway, I see Trealla as a small, compact, interesting, out of the box working Prolog; thank you .

**Comment by infradig** [comment]
Hi Peter, I don't really encourage anyone to play around with two libs yet,
still very buggy.

**Comment by infradig** [comment, marked as answer]
Normally library files are compiled into the executable to avoid path
problems, but clpb.pl & clpz.pl aren't yet because I don't want to sanction
use. I believe it's close to working, there just seems to be the one major
hurdle to overcome.

**Reply by haijinSk** [reply]
Hi Andrew, thank you for the answers. I was thinking in this direction, but I was not sure.
